metricbeat/beater/reload.go

```diff
@@ -0,0 +1,252 @@
+package beater
```

Comment:
Fully agree, that we probably should extend this to other beats, especially filebeat. Means this functionality will move to libbeat. I prefer to keep it currently in metricbeat to keep the changes as minimal as possible and do the abstraction in a second step when we add it to other beats.